src/javascripts/cookie-functions.js

```diff
@@ -0,0 +1,161 @@
+// used by the cookie banner component
+const DEFAULT_COOKIE_CONSENT = {
+  analytics: false
+}
+
+const COOKIE_CATEGORIES = {
+  _ga: 'analytics',
+  _gid: 'analytics',
+  _gat_govuk_shared: 'analytics'
+}
+
+export function getCookie (name) {
+  var nameEQ = name + '='
+  var cookies = document.cookie.split(';')
+  for (var i = 0, len = cookies.length; i < len; i++) {
+    var cookie = cookies[i]
+    while (cookie.charAt(0) === ' ') {
+      cookie = cookie.substring(1, cookie.length)
+    }
+    if (cookie.indexOf(nameEQ) === 0) {
+      return decodeURIComponent(cookie.substring(nameEQ.length))
+    }
+  }
+  return null
+}
+/*
+Cookie methods
+==============
+
+Usage:
+
+  Setting a cookie:
+  setCookie('hobnob', 'tasty', { days: 30 })
+
+  Reading a cookie:
+  getCookie('hobnob')
+
+  Deleting a cookie:
+  Cookie('hobnob', null)
+*/
+
+function Cookie (name, value, options) {
+  if (typeof value !== 'undefined') {
+    if (value === false || value === null) {
+      return setCookie(name, '', { days: -1 })
+    } else {
+      // Default expiry date of 30 days
+      if (typeof options === 'undefined') {
+        options = { days: 30 }
+      }
+      return setCookie(name, value, options)
+    }
+  } else {
+    return getCookie(name)
+  }
+}
+
+export function getConsentCookie () {
+  var consentCookie = getCookie('dm_cookies_policy')
+  var consentCookieObj
+
+  if (consentCookie) {
+    try {
+      consentCookieObj = JSON.parse(consentCookie)
+    } catch (err) {
+      return null
+    }
+
+    if (typeof consentCookieObj !== 'object' && consentCookieObj !== null) {
+      consentCookieObj = JSON.parse(consentCookieObj)
+    }
+  } else {
+    return null
+  }
+
+  return consentCookieObj
+}
+
+export function setConsentCookie (options) {
+  var cookieConsent = getConsentCookie()
+
+  if (!cookieConsent) {
+    cookieConsent = JSON.parse(JSON.stringify(DEFAULT_COOKIE_CONSENT))
+  }
+
+  for (var cookieType in options) {
+    cookieConsent[cookieType] = options[cookieType]
+
+    // Delete cookies of that type if consent being set to false
+    if (!options[cookieType]) {
+      for (var cookie in COOKIE_CATEGORIES) {
+        if (COOKIE_CATEGORIES[cookie] === cookieType) {
+          Cookie(cookie, null)
+
+          if (Cookie(cookie)) {
+            document.cookie = cookie + '=;expires=' + new Date() + ';domain=' + window.location.hostname.replace(/^www\./, '.') + ';path=/'
+          }
+        }
+      }
+    }
+  }
+
+  setCookie('dm_cookies_policy', JSON.stringify(cookieConsent), { days: 365 })
+}
+
+function checkConsentCookieCategory (cookieName, cookieCategory) {
```

Comment:
I know this isn't your change, I wonder if we can rename this function to make it a bit clearer what it does? Maybe something like `checkUserConsentForCategory` . What do you think?

Reply:
How about `checkConsentForCookieCategory`?

We might then also want to rename the function below to `checkConsentForCookie`, what do you think?

Reply:
Looking more closely at what these functions do and are used for, maybe `hasUserConsentedToCookie` and `hasUserConstentedToCookieCategory` would be better? Or is that too wordy?

Reply:
Another option 😅 : `allowSettingCookie`, `allowSettingCookieInCategory`?

Reply:
Went with `userAllowsCookie` and `userAllowsCookieCategory`.